Spreadsheet: Reuse save and rename actions

krkk opened the pull request:
*just some small cleanups* :)

- **Spreadsheet: Reuse save and rename actions**

  These parts of code were identical to their action counterparts.

- **Spreadsheet: Create rename action by using a GUI::CommonActions helper**

  Besides from reusing more parts from the code, this allows us to call the action using the F2 key. That is also the reason why we have to reassign `m_tab_context_menu_sheet_view` on tab change.

elcuco commented on Userland/Applications/Spreadsheet/SpreadsheetWidget.cpp:
```diff
@@ -357,8 +353,11 @@ void SpreadsheetWidget::setup_tabs(NonnullRefPtrVector<Sheet> new_sheets)
         change(selected_widget);
+
+        // for keyboard shortcuts and command palette
+        m_tab_context_menu_sheet_view = static_cast<SpreadsheetView&>(selected_widget);
```
Is it possible to set `m_tab_context_menu_sheet_view ` inside `change()` ?

krkk replied:
yeah, it is possible, but I don’t see a reason why we should move it there honestly.

It looks like the `change()` lambda function just moves function events from one view to another, so it looks more like a setup function (that probably should be improved) so setting the current tab there for the context menu doesn’t seem right.

krkk replied:
a small update: The `change()` function no longer exists (see #13032). Things from that function are now set immediately when making a tab.

krkk commented:
Changes since last push: rebased and resolved conflicts

AtkinsSJ commented:
This has a conflict.

krkk commented:
Changes since last push:
- rebased and resolved conflicts
- pushed one commit that fixes an issue that appeared after this rebase :p

oh and thanks @AtkinsSJ for reminding me of this PR!
